### ISSUE TITLE
您好，官方给出ERNIE预训练代码跑不通，输出一直是0it [00:00, ?it/s]，无法用其进行预训练，请问是什么原因

### ISSUE THREAD

**Issue by Cxh-001** [PaddlePaddle/ERNIE#539]
使用的是官方给出的demo目录下的pretrain/pretrain_dygraph.py 代码，无法正常输出loss。
在自己的gpu服务器和aistudio都跑过，都会出现这种问题。
输出一直是0it [00:00, ?it/s]
环境：
paddle1.7+
numpy
pyzmq==18.0.2
six==1.11.0
sklearn==0.0
sentencepiece==0.1.8
jieba==0.39
visualdl>=2.0.0b7

**Comment by oyxuan-11** [comment]
你好，请问你自己修改了哪里呢？还有麻烦贴一下完整的报错信息 log 和训练配置

**Comment by Cxh-001** [comment]
您好，我没有改动代码，按照https://github.com/PaddlePaddle/ERNIE/tree/develop/demo/pretrain的说明运行的，
我先用python3 ./demo/pretrain/make_pretrain_data.py ./demo/pretrain/input_file.txt ./data/output_file.gz --vocab ./pretrained_ernie/vocab.txt命令生成了预训练数据output_file.gz，
再用python3 -m paddle.distributed.launch ./demo/pretrain/pretrain_dygraph.py&nbsp; &nbsp;--data_dir "data/*.gz"&nbsp; &nbsp;--from_pretrained ernie-1.0&nbsp;进行训练
完整的log信息如下：
aistudio@jupyter-228947-493975:~/work/ERNIE-develop$ python3 -m paddle.distributed.launch ./demo/pretrain/pretrain_dygraph.py&nbsp; &nbsp;--data_dir "data/*.gz"&nbsp; &nbsp;--from_pretrained ernie-1.0
-----------&nbsp; Configuration Arguments -----------
cluster_node_ips: 127.0.0.1
log_dir: None
log_level: 20
node_ip: 127.0.0.1
print_config: True
selected_gpus: None
started_port: None
training_script: ./demo/pretrain/pretrain_dygraph.py
training_script_args: ['--data_dir', 'data/*.gz', '--from_pretrained', 'ernie-1.0']
use_paddlecloud: False
------------------------------------------------
INFO 2020-08-10 21:11:27,499 launch.py:210] get cluster from args:job_server:None pods:['rank:0 id:None addr:127.0.0.1 port:None visible_gpu:[] trainers:["gpu:[\'0\'] endpoint:127.0.0.1:36455 rank:0"]'] job_stage_flag:None hdfs:None
INFO 2020-08-10 21:11:27,499 utils.py:370] start trainer proc:['/opt/conda/envs/python35-paddle120-env/bin/python3', '-u', './demo/pretrain/pretrain_dygraph.py', '--data_dir', 'data/*.gz', '--from_pretrained', 'ernie-1.0'] env:{'FLAGS_selected_gpus': '0', 'PADDLE_TRAINER_ID': '0', 'PADDLE_CURRENT_ENDPOINT': '127.0.0.1:36455', 'PADDLE_TRAINERS_NUM': '1', 'PADDLE_TRAINER_ENDPOINTS': '127.0.0.1:36455'}
2020-08-10 21:11:28,721-INFO: get pretrain dir from https://ernie-github.cdn.bcebos.com/model-ernie1.0.1.tar.gz
2020-08-10 21:11:28,722-DEBUG: https://ernie-github.cdn.bcebos.com/model-ernie1.0.1.tar.gz cached in /tmp/466eabcffd6d6a83ae9cb97dd1a167bd
2020-08-10 21:11:28,733-DEBUG: regex: (\[UNK\])|(\[CLS\])|(\[SEP\])|(\[PAD\])|(\[MASK\])|([a-zA-Z0-9]+|\S)
[INFO] 2020-08-10 21:11:28,734 [pretrain_dygraph.py:&nbsp; 152]:&nbsp; &nbsp; &nbsp;read from data/train.gz
[INFO] 2020-08-10 21:11:28,734 [pretrain_dygraph.py:&nbsp; 238]:&nbsp; &nbsp; &nbsp;Apply sharding in distribution env
2020-08-10 21:11:28,742-INFO: get pretrain dir from https://ernie-github.cdn.bcebos.com/model-ernie1.0.1.tar.gz
2020-08-10 21:11:28,742-DEBUG: https://ernie-github.cdn.bcebos.com/model-ernie1.0.1.tar.gz cached in /tmp/466eabcffd6d6a83ae9cb97dd1a167bd
2020-08-10 21:11:28,743-DEBUG: init ErnieModel with config: {'attention_probs_dropout_prob': 0.1, 'hidden_act': 'relu', 'hidden_dropout_prob': 0.1, 'hidden_size': 768, 'initializer_range': 0.02, 'max_position_embeddings': 513, 'num_attention_heads': 12, 'num_hidden_layers': 12, 'type_vocab_size': 2, 'vocab_size': 18000}
W0810 21:11:28.744092&nbsp; 1911 device_context.cc:252] Please NOTE: device: 0, CUDA Capability: 70, Driver API Version: 10.1, Runtime API Version: 9.0
W0810 21:11:28.748243&nbsp; 1911 device_context.cc:260] device: 0, cuDNN Version: 7.3.
2020-08-10 21:11:31,750-INFO: loading pretrained model from /tmp/466eabcffd6d6a83ae9cb97dd1a167bd
2020-08-10 21:11:34,848-WARNING: param:pooler_heads.0.nsp.weight not set in pretrained model, skip
2020-08-10 21:11:34,848-WARNING: param:pooler_heads.0.nsp.bias not set in pretrained model, skip
0it [00:00, ?it/s]

请您帮忙看一下，谢谢！




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "FantasticSix"<notifications@github.com&gt;; 
发送时间: 2020年8月10日(星期一) 晚上9:18
收件人: "PaddlePaddle/ERNIE"<ERNIE@noreply.github.com&gt;; 
抄送: "陈晓航"<915890515@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [PaddlePaddle/ERNIE] 您好，官方给出ERNIE预训练代码跑不通，输出一直是0it [00:00, ?it/s]，无法用其进行预训练，请问是什么原因 (#539)





 
你好，请问你自己修改了哪里呢？还有麻烦贴一下完整的报错信息 log 和训练配置

**Comment by Meiyim** [comment]
你可以先检查一下训练数据的制作是否正确，有可能是产出了空数据导致pretrain hang死。
我在附件里贴了一个对预训练数据文件进行明文可视化的脚本。

[view_pretrian_gz.py.tar.gz](https://github.com/PaddlePaddle/ERNIE/files/5051105/view_pretrian_gz.py.tar.gz)

**Comment by Cxh-001** [comment]
> 你可以先检查一下训练数据的制作是否正确，有可能是产出了空数据导致pretrain hang死。
> 我在附件里贴了一个对预训练数据文件进行明文可视化的脚本。
> 
> [view_pretrian_gz.py.tar.gz](https://github.com/PaddlePaddle/ERNIE/files/5051105/view_pretrian_gz.py.tar.gz)

请问这样
![image](https://user-images.githubusercontent.com/59902846/89791375-d5916480-db55-11ea-88c7-c390e29d8197.png)是产出了空数据吗？输入文件是先自己分好词并用空格隔开再调用make_pretrain_data.py吗？或者您能否给一份样例输入文件

**Comment by Meiyim** [comment]
> 先自己分好词并用空格隔开再调用make_pretrain_data.py吗

是的，空格是用来标记MaskedLM任务中mask的粒度的(Knowledge Integration)。如果按照切词划分空格，则是`Phrase-level masking`策略；若是按照实体划分空格，则是`entity-level masking`策略（参考ernie1.0的相关paper）.

输入的样例可以参考[readme](https://github.com/PaddlePaddle/ERNIE/tree/develop/demo/pretrain).  样例文件因为容量太大，无法上传github。

**Comment by Cxh-001** [comment]
好的好的，我再尝试一下，非常感谢您！

**Comment by SuixueWang** [comment]
> 你可以先检查一下训练数据的制作是否正确，有可能是产出了空数据导致pretrain hang死。
> 我在附件里贴了一个对预训练数据文件进行明文可视化的脚本。
> 
> [view_pretrian_gz.py.tar.gz](https://github.com/PaddlePaddle/ERNIE/files/5051105/view_pretrian_gz.py.tar.gz)

你好，我也遇到这个问题：0it [00:00, ?it/s] 卡死不动；

已确保输入数据格式无空格，仅是分词后用空格划分，接着用你这个脚本来校验gz文件，输出以下报错信息，请问下这个怎么解决？

![image](https://user-images.githubusercontent.com/16064733/105285301-11ddec00-5bef-11eb-9232-42e3d31a21af.png)

**Comment by SuixueWang** [comment]
> 好的好的，我再尝试一下，非常感谢您！

你好，请问你的这个问题后来怎么解决了？是什么原因呢

**Comment by liuli996624** [comment]
你好我遇到相似的问题我是按照源代码进行运行的，但是出现了进度条卡住住的问题，可否帮忙解决一下，提下意见。 https://github.com/jerry4h/Face_Xray/blob/master/faceBlending.py
